File: EasyPC/UI/easy_pc_admin/lib/models/build_wizard_step.dart
class BuildWizardStep {
  final int stepNumber;
  final String stepName;
  final String componentType;

  BuildWizardStep({
    required this.stepNumber,
    required this.stepName,
    required this.componentType,
  });

  factory BuildWizardStep.fromJson(Map<String, dynamic> json) {
    return BuildWizardStep(
      stepNumber: json['stepNumber'] as int,
      stepName: json['stepName'] as String,
      componentType: json['componentType'] as String,
    );
  }

  Map<String, dynamic> toJson() => {
        'stepNumber': stepNumber,
        'stepName': stepName,
        'componentType': componentType,
      };
}

File: EasyPC/EasyPC.Model/BuildWizardStep.cs
namespace EasyPC.Model;

public class BuildWizardStep
{
    public int StepNumber { get; set; }
    public string StepName { get; set; } = string.Empty;
    public string ComponentType { get; set; } = string.Empty;
    public bool IsCompleted { get; set; }
    public int? SelectedComponentId { get; set; }
}
